### ISSUE TITLE
update anything from scrapbook got timeout

### ISSUE THREAD

**Issue by lipt0n** [Microsoft/vscode-cosmosdb#687]
Repro steps:
open scrapbook
connect
update something
observe !

Action: cosmosDB.executeMongoCommand
Error type: string
Error Message: Timed out executing MongoDB command "use meteor"

Version: 0.7.2
OS: linux

example:
this works:
db.timesheets.find( {
        "workers._id":"iKSHSGcw2iCrNyjxf",
        "workers.days.year":"2018",
        "workers.days.month":"06"
      })

this got timeout every time

db.timesheets.update( {
        "workers._id":"iKSHSGcw2iCrNyjxf",
        "year":"2018",
        "month":"06"
      },{
        "$set":{
          "workers.$.days.0.something":"yupy!"
        }
      })

also
db.timesheets.update( {
        "year":"2018",
        "month":"06"
      },{
        "$set":{
          "workers.0.days.0.something":"yupy!"
        }
      })

got timeout...so probably any update ?

**Comment by PrashanthCorp** [comment]
Hi Lipton, 
Thanks for trying out the extension and filing the issue! 


I ran the following two commands: 

```
db.timesheets.update( {
"workers._id":"iKSHSGcw2iCrNyjxf",
"year":"2018",
"month":"06"
},{
"$set":{
"workers.$.days.0.something":"yupy!"
}
});
```
This command gave me an error regarding the name of the field. It looks like having the $ in `workers.$.days.0.something` made it an invalid name. 


```
db.timesheets.update( {
"year":"2018",
"month":"06"
},{
"$set":{
"workers.0.days.0.something":"yupy!"
}
});
```
This command seemed to run without a timeout - I got back a result of how many documents were updated, etc. 


1. Please let me know if changing the field works. I couldn't find any links on $ being a valifd charcter for a field name. 
2. Please provide us a few more details to help me repro the last command.

**Comment by lipt0n** [comment]
workers.$.days.0.something shouldn't be invalid, this is how You address element in array matching "workers._id":"iKSHSGcw2iCrNyjxf",

**Comment by StephenWeatherford** [comment]
@lipt0n I'm guessing you're using the Cosmos emulator, right?  There's a known issue with some commands using the emulator (#508), which we hope to have fixed soon.

**Comment by PrashanthCorp** [comment]
Hi Lipton, the merged fix was published today (version 0.8.0). Hope this fixes your issue. :-)
Let us know how the fix holds up with your use-case.
